app/screens/channel/header/header.tsx

```diff
@@ -0,0 +1,157 @@
+import CustomStatusEmoji from '@components/custom_status/custom_status_emoji';
+import NavigationHeader from '@components/navigation_header';
+import {Navigation} from '@constants';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import {popTopScreen} from '@screens/navigation';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+import OtherMentionsBadge from './other_mentions_badge';
+
+import type {HeaderRightButton} from '@components/navigation_header/header';
+
+type ChannelProps = {
+    channelId: string;
+    customStatus?: UserCustomStatus;
+    isCustomStatusExpired: boolean;
+    componentId?: string;
+    displayName: string;
+    isOwnDirectMessage: boolean;
+    memberCount?: number;
+    searchTerm: string;
+    teamId: string;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    customStatusContainer: {
+        flexDirection: 'row',
+        height: 13,
+        left: Platform.select({ios: undefined, default: -2}),
+        marginTop: Platform.select({ios: undefined, default: 1}),
+    },
+    customStatusEmoji: {marginRight: 5},
+    customStatusText: {
+        alignItems: 'center',
+        height: 13,
+    },
+    subtitle: {
+        color: changeOpacity(theme.sidebarHeaderTextColor, 0.72),
+        ...typography('Body', 75),
+        lineHeight: 12,
+        marginBottom: 8,
+        marginTop: 2,
+        height: 13,
+    },
+}));
+
+const ChannelHeader = ({
+    channelId, componentId, customStatus, displayName,
+    isCustomStatusExpired, isOwnDirectMessage, memberCount,
+    searchTerm, teamId,
+}: ChannelProps) => {
+    const {formatMessage} = useIntl();
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const leftComponent = useMemo(() => {
+        if (isTablet || !channelId || !teamId) {
+            return undefined;
+        }
+
+        return (<OtherMentionsBadge channelId={channelId}/>);
+    }, [isTablet, channelId, teamId]);
+
+    const rightButtons: HeaderRightButton[] = useMemo(() => ([{
+        iconName: 'magnify',
+        onPress: () => {
+            DeviceEventEmitter.emit(Navigation.NAVIGATE_TO_TAB, {screen: 'Search', params: {searchTerm: `in: ${searchTerm}`}});
+            if (!isTablet) {
+                popTopScreen(componentId);
+            }
+        },
+    }, {
+        iconName: Platform.select({android: 'dots-vertical', default: 'dots-horizontal'}),
+        onPress: () => true,
+        buttonType: 'opacity',
+    }]), [channelId, isTablet, searchTerm]);
+
+    const onBackPress = useCallback(() => {
+        Keyboard.dismiss();
+        popTopScreen(componentId);
+    }, []);
+
+    const onTitlePress = useCallback(() => {
+        // eslint-disable-next-line no-console
+        console.log('Title Press go to Channel Info', displayName);
+    }, [channelId]);
```

Comment:
I understand this is "prepared for the future" where we will need the channelId, but if we are purist this depends on displayName. Either way, since it is a temporal debug message, I would leave the dependencies empty.